src/components/tooltip/Tooltip.js

```diff
@@ -50,32 +51,24 @@ export class Tooltip extends React.PureComponent<Props, State> {
     this._isMounted = false;
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    if (nextProps.children !== this.props.children) {
-      // If the children are different, allow them to do an initial lay out on the DOM.
-      this.setState({ isNewContentLaidOut: false });
-      this._forceUpdateAfterRAF();
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.mouseX !== this.props.mouseX) {
```

Comment:
This check makes sure to only call `_scheduleRelayout` when the cursor moves along the x axis else it'll keep on re-rendering the tooltip even if the cursor remains in the same position.

Reply:
I'm not 100% sure that this won't bring bugs. What happens if the content changes but the user doesn't move the mouse? This may happen in some components in some cases, so I would say that we can have a glitch. I'd remove this check so that we have the same behavior as wit hthe previous patch.